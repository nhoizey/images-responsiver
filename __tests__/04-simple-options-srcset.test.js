'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe('image with simple srcset options', () => {
	test('simple image', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				widthsList: [120, 180, 320],
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with only one value', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				widthsList: [320],
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with values duplicates', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				widthsList: [320, 320],
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with width superior to all values', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png" width="1280"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				widthsList: [320, 640, 1024],
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with width inferior to one value', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png" width="1024"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				widthsList: [320, 800, 1280],
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with width inferior to all values', () => {
		const content = `<!DOCTYPE html><html><body><img src="test.png" width="400"></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				widthsList: [640, 1024, 1280],
			},
		});
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
