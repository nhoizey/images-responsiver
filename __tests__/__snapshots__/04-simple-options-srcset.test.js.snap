// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image with simple srcset options simple image 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-120.png 120w, test-180.png 180w, test-320.png 320w\\"
      src=\\"test-640.png\\"
    />
  </body>
</html>
"
`;

exports[`image with simple srcset options simple image with only one value 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w\\"
      src=\\"test-640.png\\"
    />
  </body>
</html>
"
`;

exports[`image with simple srcset options simple image with values duplicates 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w\\"
      src=\\"test-640.png\\"
    />
  </body>
</html>
"
`;

exports[`image with simple srcset options simple image with width inferior to all values 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-400.png 400w\\"
      src=\\"test-640.png\\"
      width=\\"400\\"
    />
  </body>
</html>
"
`;

exports[`image with simple srcset options simple image with width inferior to one value 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w, test-800.png 800w, test-1024.png 1024w\\"
      src=\\"test-640.png\\"
      width=\\"1024\\"
    />
  </body>
</html>
"
`;

exports[`image with simple srcset options simple image with width superior to all values 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w, test-640.png 640w, test-1024.png 1024w\\"
      src=\\"test-640.png\\"
      width=\\"1280\\"
    />
  </body>
</html>
"
`;
